src/SendGrid/Helpers/Mail/SendGridMessage.cs

```diff
@@ -566,7 +566,7 @@ public void AddContents(List<Content> contents)
         }
 
         /// <summary>
-        /// Add an attachment from a stream to the email. No attachment will be added in the case that the stream cannot be read. Streams of length greater than int.MaxValue are truncated.
+        /// Add an attachment from a stream to the email. No attachment will be added in the case that the stream cannot be read or is partially read. Streams of length greater than int.MaxValue are truncated.
```

Comment:
> `Streams of length greater than int.MaxValue are truncated.`

The `Convert.ToInt32` in L586 will throw an overflow exception in that case.
In order to truncate it, that should be `(int)contentStream.Length`.